pynest/nest/tests/test_connect_arrays.py

```diff
@@ -20,55 +20,109 @@
 # along with NEST.  If not, see <http://www.gnu.org/licenses/>.
 
 import unittest
-import nest
 import numpy as np
 
+import nest
+
 nest.set_verbosity('M_WARNING')
 
 
 class TestConnectArrays(unittest.TestCase):
 
+    non_unique = np.array([1, 1, 3, 5, 4, 5, 9, 7, 2, 8], dtype=np.uint64)
+
     def setUp(self):
         nest.ResetKernel()
 
-    def test_connect_arrays(self):
-        """Connecting NumPy arrays of node IDs"""
+    def test_connect_arrays_unique(self):
+        """Connecting NumPy arrays of unique node IDs"""
         n = 10
         nest.Create('iaf_psc_alpha', n)
         sources = np.arange(1, n+1, dtype=np.uint64)
         targets = np.arange(1, n+1, dtype=np.uint64)
-        weights = np.ones(len(sources))
-        delays = np.ones(len(sources))
-        syn_model = 'static_synapse'
+        weights = 1.5
+        delays = 1.4
+
+        nest.Connect(sources, targets, syn_spec={'weight': weights, 'delay': delays})
+
+        conns = nest.GetConnections()
+
+        self.assertEqual(len(conns), n*n)
 
-        nest.Connect(sources, targets, syn_spec={'weight': weights, 'delay': delays,
-                                                 'synapse_model': syn_model})
+        for c in conns:
+            np.testing.assert_approx_equal(c.weight, weights)
+            np.testing.assert_approx_equal(c.delay, delays)
+
+    def test_connect_arrays_nonunique(self):
+        """Connecting NumPy arrays with non-unique node IDs"""
+        n = 10
+        nest.Create('iaf_psc_alpha', n)
+        sources = np.arange(1, n+1, dtype=np.uint64)
+        targets = self.non_unique
```

Comment:
It may be better to also have `n` and `np.arange(1, n+1, dtype=np.uint64)` as class variables.

Reply:
sorry, I don't have time for that ;)